dev/tools/gen_localizations.dart

```diff
@@ -157,33 +225,93 @@ String generateTranslationBundles() {
-      if (localeName == languageName)
-        continue;
-      countryCodeCount += 1;
-      final String camelCaseLocaleName = camelCase(localeName);
-      final Map<String, String> localeResources = localeToResources[localeName];
-      final String localeClassName = 'MaterialLocalization$camelCaseLocaleName';
-      final String constructor = generateConstructor(localeClassName, localeName);
-      output.writeln('');
-      output.writeln('/// The translations for ${describeLocale(localeName)} (`$localeName`).');
-      output.writeln('class $localeClassName extends $languageClassName {');
-      output.writeln(constructor);
-      for (String key in localeResources.keys) {
-        if (languageResources[key] == localeResources[key])
+    int scriptCodeCount = 0;
+    if (languageToScriptCodes.containsKey(languageName)) {
+      scriptCodeCount = languageToScriptCodes[languageName].length;
+      // Language has scriptCodes, so we need to properly fallback countries to corresponding
+      // script default values before language default values.
+      for (String scriptCode in languageToScriptCodes[languageName]) {
+        final String camelCaseScript = camelCase(languageName + '_' + scriptCode);
+        final Map<String, String> scriptResources = localeToResources[languageName + '_' + scriptCode];
+        final String scriptClassName = 'MaterialLocalization$camelCaseScript';
+        final String constructor = generateConstructor(scriptClassName, languageName + '_' + scriptCode);
+        output.writeln('');
+        output.writeln('/// The translations for ${describeLocale(languageName)} (`$languageName`).');
+        output.writeln('class $scriptClassName extends $languageClassName {');
+        output.writeln(constructor);
+        for (String key in scriptResources.keys) {
+          if (languageResources[key] == scriptResources[key])
+            continue;
+          final Map<String, dynamic> attributes = localeToResourceAttributes['en'][key];
+          output.writeln(generateGetter(key, scriptResources[key], attributes));
+        }
+        output.writeln('}');
+
+        final List<String> localeCodes = languageToLocales[languageName]..sort();
+        for (String localeName in localeCodes) {
+          if (localeName == languageName)
+            continue;
+          if (localeName == languageName + '_' + scriptCode)
+            continue;
+          if (!localeName.contains(scriptCode))
+            continue;
+          countryCodeCount += 1;
+          final String camelCaseLocaleName = camelCase(localeName);
+          final Map<String, String> localeResources = localeToResources[localeName];
+          final String localeClassName = 'MaterialLocalization$camelCaseLocaleName';
+          final String constructor = generateConstructor(localeClassName, localeName);
+          output.writeln('');
+          output.writeln('/// The translations for ${describeLocale(localeName)} (`$localeName`).');
```

Comment:
describeLocale doesn't seem to be producing the complete description when there's a script code and a country code:

```
class MaterialLocalizationZhHantHk extends MaterialLocalizationZhHant {
  /// Create an instance of the translation bundle for Chinese, using the Han script.
```